### PR TITLE
SSO Invite: Move invite status under invite details

### PR DESCRIPTION
**Pull request by agrullon95**
Related to #35278

## Proposed Changes

* These changes move the invite status underneath the invite details, allows the email/username to overflow and fixes the button overlapping issues on mobile viewport.

### Before:

Desktop:
CleanShot 2024-01-27 at 11 52 16@2x

Mobile:
CleanShot 2024-01-29 at 09 47 30@2x

### After: 

Desktop:
CleanShot 2024-01-27 at 11 57 13@2x

Mobile:

CleanShot 2024-01-29 at 09 49 42@2x

## Testing Instructions

* Checkout branch or use calypso live link
* Navigate to `http://calypso.localhost:3000/people/team/{SITE_SLUG}`
* Add/Invite a new user to your site, if you haven't already.
* Click on the new pending invite to see more details about the user and the invitation.
* Test the "Resend Invite" and "Revoke" buttons.
* The "Resend Invite" button text should change to "Invite Sent!" after a successful invite request has been made.
* The "Revoke" button should display an "Invite deleted" notice after the invite has been revoked and you should be navigated to the previous page.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/people/invite-status/index.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+					} ) }
+					onClick={ onResend }
+					busy={ requestingResend }
+				>
+					{ resendSuccess ? translate( 'Invite Sent!' ) : translate( 'Resend Invite' ) }
+				</Button>
+			) }
+			<Button
+				className="people-list-item__invite-revoke"
+				busy={ deletingInvite }
```
Comment (alshakero) [inline client/my-sites/people/invite-status/index.jsx:44]: Let's disable this one when `requestingResend` is true. Now you can resend **_and_** revoke an invite.

File: client/my-sites/people/invite-status/index.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+				'is-invite-details': type === 'invite-details',
+			} ) }
+		>
+			{ isPending && (
+				<Button
+					className={ classNames( 'people-list-item__invite-resend', {
+						'is-success': resendSuccess,
+					} ) }
+					onClick={ onResend }
+					busy={ requestingResend }
```
Comment (alshakero) [inline client/my-sites/people/invite-status/index.jsx:35]: Let's disable this one when `deletingInvite` is true. Now you can resend **_and_** revoke an invite.

**Review by alshakero — APPROVED** [review]
Tested on desktop and mobile. Works great. Found one UX issue.

**Review by crisbusquets — APPROVED** [review]
💯

File: client/my-sites/people/invite-status/index.jsx (hunk 1 above)
Comment (escapemanuele) [reply client/my-sites/people/invite-status/index.jsx:44]: I think `requestingResend` should be added to `deletingInvite`? 🤔

File: client/my-sites/people/invite-status/index.jsx (hunk 2 above)
Comment (escapemanuele) [reply client/my-sites/people/invite-status/index.jsx:35]: Same thing as before

**Review by escapemanuele — APPROVED** [review]
Let's review the `busy` field and then merge this ✌️

File: client/my-sites/people/invite-status/index.jsx (hunk 1 above)
Comment (alshakero) [reply client/my-sites/people/invite-status/index.jsx:44]: I don't think so. Making the button busy implies that the button is doing the action written on it. Disabled makes more sense IMO.

Comment (escapemanuele) [reply client/my-sites/people/invite-status/index.jsx:44]: Yes, I mistaken busy and disabled, thanks!

Comment (agrullon95) [reply client/my-sites/people/invite-status/index.jsx:44]: I agree , disabled makes sense here and will be a nice UX update. I'll quickly add this in this PR.

**Pull request merged**